src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9995,7 +9995,12 @@ void Unit::ProcSkillsAndReactives(bool isVictim, Unit* procTarget, uint32 typeMa
     if (typeMask & MELEE_BASED_TRIGGER_MASK && procTarget)
     {
         // Update skills here for players
-        if (GetTypeId() == TYPEID_PLAYER)
+        // only when you are not fighting other players or their pets/totems (pvp)
+        if (GetTypeId() == TYPEID_PLAYER &&
+                procTarget->GetTypeId() != TYPEID_PLAYER &&
+                !(procTarget->IsTotem() && procTarget->ToTotem()->GetOwner()->GetControllingPlayer()) &&
```

Comment:
I would change `GetControllingPlayer()` to just `IsPlayer()`, you probably don't want to have totems suddenly starti granting skills when their owner gets mind controlled by a boss (and I'm not aware of any player-owner pets/minions that can summon totems)

Reply:
> you probably don't want to have totems suddenly starti granting skills when their owner gets mind controlled by a boss

This won't be like that because of `GetTypeId() == TYPEID_PLAYER`. 
Anyway, I agree that `IsPlayer()` seems like a more suitable check here, but because `IsPlayer()` doesn't exist for `Unit` I had to change `GetOwner()` to `GetOwnerGUID`.

Reply:
`IsPlayer()` was ported back from master branch recently and now exists in 3.3.5 as well (as member of `Object`), you just can't see it since this PR is based on older commit

Reply:
oh, ok. Rebased to bf62901053d20da790e78394cd9b2a3129d28afa and made commit amend 9e952fde7b770d60b1732f211ff41032a2f54b37.